awspub/image.py

```diff
@@ -500,8 +459,71 @@ def create(self) -> Dict[str, _ImageInfo]:
         if self.conf["share"]:
             self._share(self.conf["share"], images)
 
+        if missing_regions:
+            logger.error("Failed to publish snapshot to all regions", extra={"missing_regions": missing_regions})
+            raise exceptions.IncompleteSnapshotSetException("Incomplete snapshot set published")
+
         return images
 
+    def _register_snapshots(self, snapshot_ids: Dict[str, str], region: str, ec2client_region: EC2Client):
```

Comment:
that seems to be very missleading. this is actually registering the AMI, not the snapshot.